python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -335,11 +344,249 @@ def write_parquet_file(context):
         self.execute_queries(sql_queries)
 
 
+class SnowflakeResource(ConfigurableResourceFactory):
```

Comment:
`ConfigurableResourceFactory[SnowflakeConnection]` will help type hinting in some places

Reply:
For a factory I also think our naming convention should be:

`UnderlyingClassResource(ConfigurableResourceFactory[UnderlingClass]`

so in this case

`SnowflakeConnectionResource(ConfigurableResourceFactory[SnowflakeConnection]`

Reply:
moved to the `ConfigurableResource` with a `get_client` method pattern